plugins/c9.ide.run.debug/debuggers/gdb/gdbdebugger.js

```diff
@@ -58,22 +61,25 @@ define(function(require, exports, module) {
                     ["autoshow", "true"]
                 ]);
             });
-            
+
             // must register ASAP, or debugger won't be ready for reconnects
             debug.registerDebugger(TYPE, plugin);
-            
+
             // writeFile root is workspace directory, unless given ~
             var shimPath = "~/.c9/bin/c9gdbshim.js";
             var shim = require("text!./shim.js");
             fs.exists(shimPath, function(exists) {
-                if (exists) return; // TODO use localfs extend when it is ready
-                fs.writeFile(shimPath, shim, "utf8", function(err) {
-                    if (err) {
-                        // unregister the debugger on error
-                        debug.unregisterDebugger(TYPE, plugin);
-                        return console.log("Error writing gdb shim: " + err);
-                    }
-                });
+                if (!exists || (settings.getNumber("project/debugger/gdb/@shim_version") || 0) < SHIM_VERSION) {
```

Comment:
Saving this in project settings will create problems for people who add their project setting to git.
Can we put version in the path (like `"~/.c9/bin/c9gdbshim1.js"`), to not store a setting, or do you think that will break something else?

Reply:
Oh, so maybe 

```
if ~/.c9/bin/c9gdbshim{x}.js doesn't exist
    remove ~/.c9/bin/c9gdbshim*
    write ~/.c9/bin/c9gdbshim{x}.js
    symlink ~/.c9/bin/c9gdbshim.js to it
```
?

Reply:
Apparently `fs.symlink` doesn't really support force-symlinking, so we'd have to check if `~/.c9/bin/c9gdbshim.js` exists, and if so, remove, then symlink. Removing directly could cause `fs.errors` to show an error if `c9gdbshim.js` doesn't exist initially per https://github.com/c9/core/blob/master/plugins/c9.fs/fs.errors.js#L43.

Thoughts?

Reply:
Instead of symlink we could create a js file as `c9gdbshim.js` with something like `require("./c9gdbshim1.js")` (or readdir and launch the latest version)

Reply:
Yes, but the catch with that is that if `c9gdbshim.js` already exists, the required versioned JS file won't overwrite it, unless we overwrite every time (if I interpreted that correctly?)

Also using versioned JS files only might make it a pain for scripts and plugins that require the shim to either update the filename every time the shim is updated or do readdir or the like which might not be ideal?

Reply:
Isn't that the same with symlink?

I am proposing only to change the last step https://github.com/c9/core/pull/455#discussion_r141496069
```
if ~/.c9/bin/c9gdbshim{x}.js doesn't exist
    remove ~/.c9/bin/c9gdbshim*
    write ~/.c9/bin/c9gdbshim{x}.js
    write ~/.c9/bin/c9gdbshim.js which contains require('c9gdbshim{x}.js')
```

this should work same as symlink, because [shim.js](https://github.com/c9/core/blob/master/plugins/c9.ide.run.debug/debuggers/gdb/shim.js) doesn't export anything

Reply:
@nightwing how does 7b3fe8b look?

Reply:
@kzidane looks good, will merge this today.